core/src/main/scala/org/apache/spark/executor/CoarseGrainedExecutorBackend.scala

```diff
@@ -80,9 +78,8 @@ private[spark] class CoarseGrainedExecutorBackend(
   }
 
   override def receive: PartialFunction[Any, Unit] = {
-    case RegisteredExecutor =>
+    case RegisteredExecutor(hostname) =>
```

Comment:
note: I'm having to change this because the client doesn't know which hostname the client connection will be created on when the register message is sent. So the server replies it back to the client, so that both agree about what's the actual address being used.